Unable to update properties in v3.0.0

### What happened?

I have seen this issue across multiple stacks when updating properties. Most recent error message below:

m2022/09/14 09:08:06 [ERROR] could not read property version: OperationID=5601cd44-6843-497f-b2e9-bb6c7e33651e PAPI=resourcePropertyUpdate property_version=0 property_name=prp_834888 error="fetching property version: struct validation: PropertyVersion: cannot be blank." contract_id=ctr_M-1MEE94H group_id=grp_156959

I have checked in the control center, and the current property version is 1.

After exporting the stack to JSON, I see the properties have `readVersion` set to zero after each update. A similar issue was reported previously: #50 

### Steps to reproduce

To reproduce this bug:
1. Create a Pulumi stack with a property in it
2. Deploy the stack
3. Make an update to the property's rules
4. Deploy the stack again

This is all I did, it should be enough to show the error.

### Expected Behavior

The property to update, as specified.

### Actual Behavior

Error

### Output of `pulumi about`

```
CLI          
Version      3.31.0
Go Version   go1.17.9
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host     
OS       darwin
Version  12.5.1
Arch     x86_64

This project is written in nodejs (/Users/lpopp/.nvm/versions/node/v14.18.2/bin/node v14.18.2)

Current Stack: gcd-akamai-cdn-eu-dev

TYPE                                            URN
pulumi:pulumi:Stack                             urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:pulumi:Stack::vdp-gcd-akamai-cdn-gcd-akamai-cdn-eu-dev
pulumi:providers:aws                            urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:aws::aws-eu-west-1
pulumi:providers:aws                            urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:aws::aws-ap-southeast-1
pulumi:providers:aws                            urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:aws::aws-us-east-1
aws:secretsmanager/secret:Secret                urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::aws:secretsmanager/secret:Secret::akamai-dev-gb-sport-key
pulumi:providers:akamai                         urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:akamai::dev-akamai-provider
pulumi:providers:random                         urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:random::default_4_6_0
akamai:index/edgeHostName:EdgeHostName          urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::akamai:index/edgeHostName:EdgeHostName::dev-eu-live-hn-gb-sport-live-dev.akamaized.net
random:index/randomPassword:RandomPassword      urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::random:index/randomPassword:RandomPassword::dev-gb-sport-key
aws:secretsmanager/secretVersion:SecretVersion  urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::aws:secretsmanager/secretVersion:SecretVersion::akamai-dev-gb-sport-key
akamai:index/cpCode:CpCode                      urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::akamai:index/cpCode:CpCode::dev-eu-live-cp
akamai:index/property:Property                  urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::akamai:index/property:Property::dev-eu-live-prop


Found no pending operations associated with gcd-akamai-cdn-eu-dev

Backend        
Name           blvl184611m
URL            s3://dev-pulumi-states
User           lpopp
Organizations  
```

### Additional context

Is it possible we are missing something in our configuration somehow?

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

Hi @guineveresaenger ,

I've finally attempted to upgrade to the 4.0.0 version, but I am still seing the same issue in this version. I am seeing the same issue as before

pulumi:pulumi:Stack <stackname> running 2023/01/17 10:00:52 [ERROR] could not read property version: OperationID=36dc529e-5ca5-4bd4-9e6c-d5f84abd4e4d PAPI=resourcePropertyUpdate property_name=<propertyname> contract_id=<contractid> group_id=<groupid> property_version=0 error="fetching property version: struct validation: PropertyVersion: cannot be blank."

This issue is not seen in version 2.9.1, which I think had a [fix](https://github.com/pulumi/pulumi-akamai/issues/101) for this. For me that problem resurfaced in the following releases.

--- a/examples/examples_nodejs_test.go
+++ b/examples/examples_nodejs_test.go
@@ -19,6 +19,20 @@ func TestAccEdgeHostnameTs(t *testing.T) {
 	integration.ProgramTest(t, &test)
 }
 
+func TestPropertyUpdate(t *testing.T) {
+	dir := path.Join(getCwd(t), "property", "ts")
+	test := getJSBaseOptions(t).
+		With(integration.ProgramTestOptions{
+			Dir: dir,
+			EditDirs: []integration.EditDir{{
+				Dir:             path.Join(dir, "update"),
+				ExpectNoChanges: false,
+				Additive:        true,
+			}},
+		})
+	integration.ProgramTest(t, &test)
+}
+
 func getJSBaseOptions(t *testing.T) integration.ProgramTestOptions {
 	base := getBaseOptions(t)
 	baseJS := base.With(integration.ProgramTestOptions{

--- a/examples/property/ts/Pulumi.yaml
+++ b/examples/property/ts/Pulumi.yaml
@@ -1,0 +1,3 @@
+name: akamai-property-ts
+runtime: nodejs
+description: A minimal TypeScript Pulumi program to create an Akamai Property

--- a/examples/property/ts/index.ts
+++ b/examples/property/ts/index.ts
@@ -1,0 +1,18 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as akamai from "@pulumi/akamai";
+
+const groupId = "grp_201328";
+const productId = "prd_Fresca";
+const contractId = akamai
+  .getContract({
+    groupId: groupId,
+  })
+  .then((x) => x.id);
+
+new akamai.Property("test", {
+  productId: productId,
+  contractId: contractId,
+  groupId: groupId,
+  hostnames: [],
+  ruleFormat: "v2020-03-04",
+});

--- a/examples/property/ts/package.json
+++ b/examples/property/ts/package.json
@@ -1,0 +1,9 @@
+{
+    "name": "akamai-property-ts",
+    "devDependencies": {
+        "@types/node": "^10.0.0"
+    },
+    "dependencies": {
+        "@pulumi/pulumi": "^3.0.0"
+    }
+}

--- a/examples/property/ts/tsconfig.json
+++ b/examples/property/ts/tsconfig.json
@@ -1,0 +1,18 @@
+{
+    "compilerOptions": {
+        "strict": true,
+        "outDir": "bin",
+        "target": "es2016",
+        "module": "commonjs",
+        "moduleResolution": "node",
+        "sourceMap": true,
+        "experimentalDecorators": true,
+        "pretty": true,
+        "noFallthroughCasesInSwitch": true,
+        "noImplicitReturns": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.ts"
+    ]
+}

--- a/examples/property/ts/update/index.ts
+++ b/examples/property/ts/update/index.ts
@@ -1,0 +1,23 @@
+import * as akamai from "@pulumi/akamai";
+
+const groupId = "grp_201328";
+const productId = "prd_Fresca";
+const contractId = akamai
+  .getContract({
+    groupId: groupId,
+  })
+  .then((x) => x.id);
+
+new akamai.Property("test", {
+  productId: productId,
+  contractId: contractId,
+  groupId: groupId,
+  hostnames: [
+    {
+      cnameFrom: "example.com",
+      cnameTo: "example.com.edgekey.net",
+      certProvisioningType: "DEFAULT",
+    },
+  ],
+  ruleFormat: "v2020-03-04",
+});

--- a/provider/resources.go
+++ b/provider/resources.go
@@ -33,6 +33,7 @@ import (
 	"github.com/pulumi/pulumi-akamai/provider/v4/pkg/version"
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
 	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
 )
 
@@ -446,6 +447,19 @@ func Provider() tfbridge.ProviderInfo {
 		makeResource(mainMod, "Property"), propertiesMod, mainMod, &tfbridge.ResourceInfo{
 			Docs: &tfbridge.DocInfo{
 				Source: "property.md",
+			},
+			Fields: map[string]*tfbridge.SchemaInfo{
+				"read_version": {
+					Transform: func(v resource.PropertyValue) (resource.PropertyValue, error) {
+						// Read version needs to be populated and non-zero on updates, but
+						// for some reason isn't. See
+						// https://github.com/pulumi/pulumi-akamai/issues/107
+						if v.IsNull() || (v.IsNumber() && v.NumberValue() == 0) {
+							v = resource.NewNumberProperty(1)
+						}
+						return v, nil
+					},
+				},
 			},
 		})
 	prov.RenameResourceWithAlias("akamai_property_activation", makeResource(propertiesMod, "PropertyActivation"),

